relay-server/src/services/processor.rs

```diff
@@ -608,29 +608,37 @@ impl ProcessingExtractedMetrics {
     }
 
     /// Extends the contained project metrics.
-    pub fn extend_project_metrics(
+    pub fn extend_project_metrics<I>(
```

Comment:
nit: now that you changed this method, I would generalize it to avoid the duplication of the `Keep` logic.

Reply:
Looked into this and it turns out this ends up adding more code. Let's keep it this way until there's more shared code.